pyomo/contrib/mindtpy/nlp_solve.py

```diff
@@ -156,8 +158,20 @@ def handle_NLP_subproblem_optimal(fixed_nlp, solve_data, config):
                 solve_data.num_no_good_cuts_added.update(
                     {solve_data.LB: len(solve_data.mip.MindtPy_utils.MindtPy_linear_cuts.integer_cuts)})
 
-    # Add the linear cut
-    if config.strategy == 'OA':
+        # add obj increasing constraint for feas_pump
+        if feas_pump:
+            if solve_data.mip.MindtPy_utils.MindtPy_linear_cuts.find_component('increasing_objective_cut') is not None:
+                solve_data.mip.MindtPy_utils.MindtPy_linear_cuts.del_component(
+                    'increasing_objective_cut')
+            if main_objective.sense == minimize:
+                solve_data.mip.MindtPy_utils.MindtPy_linear_cuts.increasing_objective_cut = Constraint(expr=solve_data.mip.MindtPy_utils.objective_value
+                                                                                                       <= solve_data.UB - config.feas_pump_delta*min(1e-4, abs(solve_data.UB)))
+            else:
+                solve_data.mip.MindtPy_utils.MindtPy_linear_cuts.increasing_objective_cut = Constraint(expr=solve_data.mip.MindtPy_utils.objective_value
+                                                                                                       >= solve_data.LB + config.feas_pump_delta*min(1e-4, abs(solve_data.LB)))
+
+            # Add the linear cut
+    if config.strategy == 'OA' or feas_pump:
```

Comment:
Just a general question. We could potentially add affine cuts to the feaspump (I don't know if that is intelligent). If that is the case how should the code look like?
In line 189 below the comment needs to be updated to `no_good_cuts`.

Reply:
Add `add_affine_cuts` option